### ISSUE TITLE
Updated From docker image bitnami/joomla 3.9.19 to 3.9.20 will not work with my named volume. Nor will bitnami/joomla: 3.10.xx or 4.0 work.

### ISSUE THREAD

**Issue by gigantor2** [bitnami/containers#3974]
### Name and Version

**Images** bitnami/joomla:3.9.20-3.9.28, 3.10.xx, and 4.0-4.1.5 updates fail when attached to my "named volume."

### What steps will reproduce the bug?

1. run docker image 3.9.20 or above with named volume created with a docker image bitnami/joomla:3.9.13-3.9.19 or below in compose file. 
2. (Note: My bitnami/joomla application had been updated from the bitnami/joomla administrative backend from 3.10.10 to bitnami/joomla:4.1.5.  (Note my reference to 4.1.5 **application** vs. (3.9.19, 3.10.xx,4.0-4.1.5) docker **image**
3. docker bitnami/joomla **image** 3.9.20 uses php 7.4 or 8.0 and above. however, it fails with errors when connected to my named volume via my compose file.
4. my version docker image 3.9.13-docker bitnami/joomla **image** and the upgraded bitnami **image** 3.9.19 (suceed/work) but use php 7.3.xx. 
5. Because bitnami/joomla image 3.9.xx use php 7.3.xx are coming to end of life, I need a docker image that has php 7.4 or above and supports my named volume (which was created with a bitnami/joomla:3.9.13) with the docker image. Also, I need the docker image that has php 7.4 or 8.0, because my newly upgraded joomla application (4.1.5) needs php 7.4 - 8.0 and above, to run my new 4.1.5 joomla extensions.

### What is the expected behavior?

I expect the website to load as it has done in the past with bitnami/joomla images 3.9.13-3.9.19 using my named volume seemlessly. Note my site loads perfectly from the compose file with my named volume using bitnami/joomla:3.9.19 docker image. 

### What do you see instead?

 Cannot load modules/libphp7.so into server"
l also, ran Docker image Joomla:3.10.10 again tonight attempting to map my volumes and got these errors:
```
mariadb_1  | mkdir: cannot create directory '/bitnami/mariadb/data': Permission denied
joomla_1   | realpath: /bitnami/apache/conf: No such file or directory
joomla_1   | grep: /opt/bitnami/joomla/configuration.php: No such file or directory
debian-11_joomla_1 exited with code 2
```
NOTE: EVERYTHING WORKS PERFECTLY IN WITH JOOMLA 3.9.19 IMAGE AND BELOW, WITH MARIADB 10.6 AND BELOW.

### Additional information

I am not running this application in a cloud environment yet, it is running on my laptop. I'm trying to get it working as well as possible before going to a hosting provider and going live in a kubernetes  environment. However, from my experience with the "minikube" environment, if it won't work in docker, it's probably is not going to work in kubernetes.

**Also, I should note that when I run bitnami/joomla:3.9.20 or above without my volume it runs perfectly. However, I get a brand new shiny site without any of my data. Also, when I run bitnami/joomla:4 or above I am required to contact a cloud provider to get my credentials. This is not good when you are running the application locally without a cloud provider. I did see where it indicated you could ssh into the server to get the credentials, but I did not try that approach since a new site without my data would do me no good.

**Comment by fmulero** [comment]
Joomla is a community-supported solution. This means that the Bitnami team is not actively working on new features/improvements nor providing support through GitHub Issues for this Container. Any new issue will stay open for 20 days to allow the community to contribute, after 15 days without activity the issue will be marked as stale being closed after 5 days.

The Bitnami team will review any PR that is created, feel free to create a PR if you find any issue or want to implement a new feature.

**Comment by gigantor2** [comment]
Greetings fmulero:
I am not a developer. I need assistance making these future containers backward compatible with my my data volumes. I read the requirements for making a PR. The requirements are overly burdensome for an individual who is not a code developer, but is using this product regularly. 

I should not have to create teams of developers, and become a member of certain teams/groups to get assistance with an issue affecting my project, and to get current code to work with my volumes. 

 **Why would the developers of this container not create a way for all users of this container to have backward compatibility? 

**Why would the developers of this container leave users of ten years in a situation where php 7.4 is required, but not available to the public in an image, or from the php repository that is backward compatible with previously created volumes.  There is only one repository "Ondre" for php  and 7.4 and 8.0+ was not available. In addition, his repository was broke, and, not providing php 7.4 and 8.0 for this image bitnami/joomla:3.9.19. 

**Numerous users are complaining online and trying to find "work arounds" to solve this problem. I am left with a container without update capabilities, and no access to php updates that load without errors and broken packages. Yet I am required, and receive warnings on my site that this version of php (php 7.3.12) will be reaching "end of life" in at the end of 2022. Also, php (7.4) will reach "end of life" in 2023.

**Also, why would the new owners, (VmWare ) allow this to happen? 

This is certainly bad for VmWare's brand, and, is totally unnecessary. Users are being informed they need php 7.4-8.0 and are not given a way to get it and use their previously created volumes. Many user have numerous websites that depend on bitnami/joomla and their volumes that are attached with their data.

Look, I do not want to make this a big deal, or a public Internet-Wide Issue, but I need someone to help me without all of these bureaucratic development Procedures. Myself, as well as many others, including many developers are in the same situation...left  "high and dry." 

I am not a programmer, nor do I understand the intricacy of this developer's code, but I'm sure he could make this right very easily, by at least, sharing and explaining what needs to be done. For instance, if I need to move apache files, or map another directory, just give me some direction on how to make this work. 

In addition, I am not familiar with creating a PR and the intricate knowledge of this container's code to participate in this manner. **Had I known this was going to be an issue, I would have never switched to bitnami joomla from the original joomla.**

**Hopefully, this can be resolved without involvement by the VmWare Corporation and the Internet community at large.**

Thanks again, in advance for any help your support team can provide.

**Comment by fmulero** [comment]
Hi @gigantor2 

Version 3.9.20-debian-10-r0 was release on July of 2020 with [several changes](https://github.com/bitnami/bitnami-docker-joomla/commit/2b8d630c557e7268650b7cbb0f8efcda680c2b76#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5) Including some breaking changes like the use of non-root container. Also as you can see in the changes that there is no relevant change in PHP version is using 7.3

In the readme file there is a specific section for [version 3.9.20](https://github.com/bitnami/containers/tree/main/bitnami/joomla#3920-debian-10-r0) where we encourage to export your site and import it in the new container. Please take a look to the documentation.

Please keep you updated regularly to avoid security issues, you could be affected by important issues like [Log4Shell](https://www.cve.org/CVERecord?id=CVE-2021-44228) appeared in 2021

**Comment by gigantor2** [comment]
Greetings fmulero:

I have been up all night and I finally figured this out about an hour ago. The answer was right in front of my face, and, I have executed these actions numerous times in the past. I can't believe I missed the solution. I did read an article on the Internet that got me going in the right direction regarding exporting my site and importing it into the new container. That little information was all I needed and then it all "clicked."

Anyway, I have at least succeeded in installing bitnami/joomla:3.10.11 which installs php 8.0. I feel pretty confident that I will be able to do the same with joomla:4.0. This at least allows me to run extensions that depend on php 8.0. I guess I got a little panicked and paranoid at the thought of being stuck with php 7.3.

Thanks again for following up with me on this issue, and I apologize for being so long winded in my responses, but I was just getting stressed and needed to "blow off some steam." I will update my other requests for assistance at other forums so the joomla community understands that this all ended on a positive note, and that the answer was documented, but I failed to read the appropriate documentation.

I definitely have to stay updated more regular, but I've been so focused trying build the site, I didn't make updating a priority until those big red signs complaining about php 7.3.12 being out of date got my attention. In addition, after updating to joomla:4.1.5 and several extensions not working because I was lacking php 8.0, I finally got  worried.

Thanks again, have a great day.
